Core/GDCore/Project/ObjectGroupsContainer.cpp

```diff
@@ -4,101 +4,147 @@
  * reserved. This project is released under the MIT License.
  */
 
-#include "GDCore/Project/ObjectGroupsContainer.h"
+#include "ObjectGroupsContainer.h"
+
 #include "GDCore/Project/ObjectGroup.h"
 #include "GDCore/Serialization/SerializerElement.h"
 #include "GDCore/String.h"
 
 namespace gd {
+gd::ObjectGroup ObjectGroupsContainer::badGroup;
+
+ObjectGroupsContainer::ObjectGroupsContainer() {}
+
+ObjectGroupsContainer::ObjectGroupsContainer(
+    const ObjectGroupsContainer& other) {
+  Init(other);
+}
+
+ObjectGroupsContainer& ObjectGroupsContainer::operator=(
+    const ObjectGroupsContainer& other) {
+  if (this != &other) Init(other);
+
+  return *this;
+}
+
+void ObjectGroupsContainer::Init(const ObjectGroupsContainer& other) {
+  objectGroups.clear();
+  for (auto& it : other.objectGroups) {
+    objectGroups.push_back(it->Clone());
```

Comment:
```suggestion
    objectGroups.push_back(gd::make_unique<gd::ObjectGroup>(*it));
```

(which will call the default copy constructor, and will return a unique_ptr to it).